### PR TITLE
Epic: create and join rooms - Start Chat

### PR DESCRIPTION
**Pull request by flescio**
Close https://github.com/vector-im/element-x-ios/issues/592

Add the button for starting the Star Chat flow, the button present the screen with only UI, more login in different tasks

### CODE REVIEW

**Review by pixlwave — COMMENTED** [review]
Nice, this is really great first PR to the project 🙌 I see you found all of the form style modifiers. Hope they were self explanatory enough to use.

I'm sure you've noticed Unit tests need implementing, its probably worth pointing out UI tests are the same. Those run nightly so its down to us to remember to check them locally first. Various other comments inline.

One of our aims with the Element X project is to avoid lots of warnings popping up in the sidebar. Ideally if we could avoid merging lots of `// TODO:` comments this would be better for everyone. I was chatting to Stefan about maybe doing something like `// ISSUE:123` instead so that they aren't warnings, but easily searchable and have an associated issue so everyone knows what is needed. Might bring this up on Friday for everyone to chat about :)

File: ElementX/Sources/Screens/StartChat/View/UserSuggested.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        HStack(spacing: 13) {
+            LoadableAvatarImage(url: user.avatarURL,
+                                name: user.displayName,
+                                contentID: user.userId,
+                                avatarSize: .user(on: .home),
+                                imageProvider: imageProvider)
+                .accessibilityHidden(true)
+            
+            VStack(alignment: .leading, spacing: 4) {
+                Text(user.displayName ?? "")
```
[the remark this answers is not in the document]
Comment (flescio) [reply ElementX/Sources/Screens/StartChat/View/UserSuggested.swift:33]: Sorry it's not very clear for me, if there isn't a display name, you suggest to put the user ID in the top row, with primary color, and hide it from the bottom row. I got it right?

File: ElementX/Sources/Application/AppSettings.swift [hunk 2]
```diff
@@ -141,6 +142,11 @@ final class AppSettings: ObservableObject {
     @UserSetting(key: UserDefaultsKeys.shouldCollapseRoomStateEvents.rawValue, defaultValue: true, persistIn: nil)
     var shouldCollapseRoomStateEvents
 
+    // MARK: - Start Chat
+    
+    @UserSetting(key: UserDefaultsKeys.showStartChatFlow.rawValue, defaultValue: false, persistIn: store)
+    var showStartChatFlow
```
[the remark this answers is not in the document]
Comment (flescio) [reply ElementX/Sources/Application/AppSettings.swift:148]: ok, glad you pointed that out, my main concern to raise was about this, how we clearly separate feature flags

File: ElementX/Sources/Screens/StartChat/View/UserSuggested.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (pixlwave) [reply ElementX/Sources/Screens/StartChat/View/UserSuggested.swift:33]: Sorry, my comment was totally unclear, but yes you interpreted that correctly 👍
Worth double checking with design that this idea makes sense, I just wanted to highlight that not having a display name is a case we need to acknowledge and handle.

File: ElementX/Sources/Application/AppSettings.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (flescio) [reply ElementX/Sources/Application/AppSettings.swift:148]: Non persisting it's probably not the best experience for testers though

File: ElementX/Sources/Screens/StartChat/View/UserSuggested.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (flescio) [reply ElementX/Sources/Screens/StartChat/View/UserSuggested.swift]: Ok thanks, I'll like the idea but I'll discuss with design then, they have the last word

File: ElementX/Sources/Application/AppSettings.swift [hunk 3]
```diff
@@ -140,7 +141,7 @@ final class AppSettings: ObservableObject {
     
     @UserSetting(key: UserDefaultsKeys.shouldCollapseRoomStateEvents.rawValue, defaultValue: true, persistIn: nil)
     var shouldCollapseRoomStateEvents
-
+
```
Comment (alfogrillo) [inline ElementX/Sources/Application/AppSettings.swift:144]:
```suggestion

```

**Review by alfogrillo — APPROVED** [review]
LGTM! 🎉

File: ElementX/Sources/Application/AppSettings.swift [hunk 4]
```diff
@@ -159,4 +153,11 @@ final class AppSettings: ObservableObject {
     // MARK: - Other
     
     let permalinkBaseURL = URL(staticString: "https://matrix.to")
+    
+    // MARK: - Feature Flags
```
Comment (pixlwave) [inline ElementX/Sources/Application/AppSettings.swift:157]: 👍

File: ElementX/Sources/Screens/StartChat/View/StartChatScreen.swift [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        .searchable(text: $context.searchQuery, placement: .navigationBarDrawer(displayMode: .always), prompt: ElementL10n.searchForSomeone)
+    }
+    
+    private var createRoomSection: some View {
+        Section {
+            Button(action: createRoom) {
+                Label(ElementL10n.createARoom, systemImage: "person.3")
+            }
+            .buttonStyle(FormButtonStyle(accessory: .navigationLink))
+            .accessibilityIdentifier("createARoomButton")
```
[the remark this answers is not in the document]
Comment (pixlwave) [reply ElementX/Sources/Screens/StartChat/View/StartChatScreen.swift]: This same comment applies to all the `.accessibilityIdentifier`s :)

File: UITests/Sources/StartChatScreenUITests.swift [hunk 6]
```diff
@@ -18,25 +18,8 @@ import ElementX
...
-        
-        let title = app.staticTexts["title"]
-        XCTAssert(title.exists)
-        
-        XCTAssertEqual(title.label, "Privacy warning")
-
-        app.assertScreenshot(.simpleUpgrade)
+    func testStartChatScreen() {
+        let app = Application.launch(.startChat)
+        app.assertScreenshot(.startChat)
```
Comment (pixlwave) [inline UITests/Sources/StartChatScreenUITests.swift:23]: There are no snapshots included in the PR to accompany this.

**Comment by pixlwave** [comment]
Super, screenshots look 👌. The unit tests are failing due to not having run SwiftFormat. I think you need to run the project setup script to get this to run on commit.
```
swift run tools setup-project
```

Then in comparison to EI we have more checks to look at. So please look at fixing the SwiftLint warnings and SonarCloud code smells that would be awesome.

**Review by pixlwave — APPROVED** [review]
LGTM 😎

**Pull request merged**